source/server/admin/prometheus_stats.cc

```diff
@@ -284,8 +316,14 @@ uint64_t PrometheusStatsFormatter::statsAsPrometheus(
   metric_name_count += outputStatType<Stats::TextReadout>(
       response, params, text_readouts, generateTextReadoutOutput, "gauge", custom_namespaces);
 
-  metric_name_count += outputStatType<Stats::ParentHistogram>(
-      response, params, histograms, generateHistogramOutput, "histogram", custom_namespaces);
+  if (params.histogram_emit_mode_ & Utility::HistogramEmitMode::Summary) {
```

Comment:
If you emit both types, I believe this will currently have duplicates for `_sum` and `_count`, which is invalid in the exposition format. I think it's probably not worth supporting both at the same time; just make it either summary or histograms.